ansible/cluster.yml

```diff
@@ -25,10 +25,51 @@
     - nodes
   sudo: yes
   roles:
-    - flannel
+    - { role: flannel, when: networking == 'flannel' }
   tags:
     - flannel
 
+# install opencontrail
+- hosts:
+    - masters[0]
+  sudo: yes
+  roles:
+    - { role: opencontrail-kmod, when: networking == 'opencontrail' }
+  tags:
+    - opencontrail-kmod
+
+- hosts:
+    - masters
+  gather_facts: yes
+  sudo: yes
+  tags:
+    - opencontrail
+    - opencontrail-gateway
```

Comment:
gather_facts is only disabled for the pre-ansible role.  It runs before the next role (or it sure better be!)